src/app/Exec/lib/backend_proxy.js

```diff
@@ -0,0 +1,38 @@
+"use strict";
+
+const { BackendProxy: BackendProxyBase } = require("backend");
+const DirectBackend = require("./backends/direct/index");
+const JenkinsBackend = require("./backends/jenkins/index");
+const Backend = require("./types/backend");
+
+const BackendTypes = {
+    direct: DirectBackend,
+    jenkins: JenkinsBackend
+};
+
+class BackendProxy extends BackendProxyBase {
+    constructor() {
+        super(Backend);
+    }
+
+    createExecutor(backend, data = null) {
+        const backendType = this.getBackend(backend).backend.backendType;
+        if (Object.keys(this.executorClasses).indexOf(backendType) > -1) {
+            return new this.executorClasses[backendType](data);
+        }
+
+        throw Error(`Cannot find executor for backend '${backend}'`);
+    }
+
+    async start(config = {}, executorClasses, ...args) {
```

Comment:
Why not have executorClasses part of config?
`this.executorClasses = Object.assign({}, config.executorClasses);`